storage/remote/queue_manager.go

```diff
@@ -629,13 +629,13 @@ func processExternalLabels(ls labels.Labels, externalLabels labels.Labels) label
 			j++
 		}
 	}
-	for ; i < len(ls); i++ {
-		result = append(result, labels.Label{
-			Name:  ls[i].Name,
-			Value: ls[i].Value,
-		})
+
+	if i < len(ls) {
```

Comment:
I think this block could actually just be:

```
result = append(result, ls[i:]...)
result = append(result, externalLabels[j:]...)
```

Reply:
Thx,  code updated, review please again.